tests/stdlib/tasynchttpserver_transferencoding.nim

```diff
@@ -1,42 +1,40 @@
+discard """
+  matrix: "--gc:arc --threads:on; --gc:arc --threads:on -d:danger; --threads:on"
+"""
+
 import httpclient, asynchttpserver, asyncdispatch, asyncfutures
 import net
 
-import std/asyncnet
 import std/nativesockets
+import std/threadpool
 
-const postBegin = """
-POST / HTTP/1.1
-Transfer-Encoding:chunked
-
-"""
-
-template genTest(input, expected) =
+template genTest(input, expected: string) =
   var sanity = false
-  proc handler(request: Request) {.async.} =
+  proc my_handler(request: Request) {.async.} =
       doAssert(request.body == expected)
       doAssert(request.headers.hasKey("Transfer-Encoding"))
-      doAssert(not request.headers.hasKey("Content-Length"))
+      # TODO: This assert depends on client-side code, not server-side code
+      # Currently, the Nim Http Client does not support Transfer-Encoding-style
+      # requests
+      # doAssert(not request.headers.hasKey("Content-Length"))
       sanity = true
       await request.respond(Http200, "Good")
 
-  proc runSleepLoop(server: AsyncHttpServer) {.async.} = 
-    server.listen(Port(0))
-    proc wrapper() = 
-      waitFor server.acceptRequest(handler)
-    asyncdispatch.callSoon wrapper
+  proc send_request(server: AsyncHttpServer): Future[AsyncResponse] {.async.} =
+    let client = newAsyncHttpClient()
+    let headers = newHttpHeaders({"Transfer-Encoding": "chunked"})
+    let  clientResponse = await client.request("http://localhost:64123/", body=input, headers=headers, httpMethod=HttpPost)
+    server.close()
+    return clientResponse
 
-  let server = newAsyncHttpServer()
-  waitFor runSleepLoop(server)
-  let data = postBegin & input
-  var socket = newSocket()
-  socket.connect("127.0.0.1", server.getPort)
-  socket.send(data)
-  waitFor sleepAsync(10)
-  socket.close()
-  server.close()
+  proc run_server(): void =
+    let server = newAsyncHttpServer()
+    discard server.serve(Port(64123), my_handler)
```

Comment:
why hardcode port? refs https://github.com/timotheecour/Nim/issues/534#issuecomment-821904874

Reply:
Good catch - sloppy old code. Fixed it.